sql: fix race in TestScanBatches

RaduBerinde opened the pull request:
Fixing a race caused by `row.TestingSetKVBatchSize` by instantiating a separate
server for each batch size and setting it before the server starts.

Fixes #53002.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53045)
<!-- Reviewable:end -->

asubiotto reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> how long does this test take to run now?

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53045#-:-MF6QkmCDrov6VE--q_W:bu0trzs)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Surprisingly, not much difference, I see 1.4s instead of 1.1s.

RaduBerinde committed:
sql: fix race in TestScanBatches

Fixing a race caused by `row.TestingSetKVBatchSize` by instantiating a separate
server for each batch size and setting it before the server starts.

Fixes #53002.

Release note: None

RaduBerinde commented:
bors r+
